pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -638,6 +638,12 @@ private static void appendTime(StringBuilder sb, int hours, int minutes, int sec
     if (needZeros > 0) {
       sb.insert(len, ZEROS, 0, needZeros);
     }
+
+    int end = sb.length() - 1;
+    while (sb.charAt(end) == '0') {
+      sb.deleteCharAt(end);
+      end--;
```

Comment:
hm, in that `00.0` case, it will get caught here:

```
if (nanos == 0) {
  return;
}
```

and not even the dot will be there, which conforms to a `YYYY-MM-DD hh:mm:ss` format for the without timezone case. I will add the test case for this though.

PS `for(...)` I had that earlier and after sleeping on it.. I changed it to a while loop since it read a bit better, less mental hoops to jump through.

```
for (int end = sb.length() - 1; sb.charAt(end) == '0'; end-- ) {
  sb.deleteCharAt(end);
}
```
I preferred the while loop after comparing the two. Thoughts?